cuckoo/cuckoo_main.py

```diff
@@ -1305,19 +1305,56 @@ def _parse_dll(self, kwargs: Dict[str, Any], task_options: List[str], parent_sec
             exports_available.append("DllRegisterServer")
 
         max_dll_exports = self.config.get("max_dll_exports_exec", 5)
-        task_options.append(f"function={'|'.join(exports_available[:max_dll_exports])}")
+
+        # If the number of available exports is greater than the maximum number of
+        # exports that we want to run, we will be prioritizing by the following:
+        # 1. well known exports (dllRegisterServer, etc)
+        # 2. first exports (10%)
+        # 3. last exports (10%)
+        # 4. least common exports (80% - 2 exports for DllRegisterServer and DllMain)
+        if len(exports_available) > max_dll_exports:
+            ten_percent_of_exports = ceil(max_dll_exports * 0.1)
+
+            # add well-known exports
+            exports_to_run.extend(["DllMain", "DllRegisterServer"])
```

Comment:
What if they are not part of exports_available, should you make sure to only add those that are part of the other list?